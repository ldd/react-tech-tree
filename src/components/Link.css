.Link-Container {
  position: absolute;
  top: 0px;
  left: 0px;
  width: 100%;
  height: 100%;
}
.Link {
  fill: none;
  stroke: gray;
  stroke-width: 6px;
}
