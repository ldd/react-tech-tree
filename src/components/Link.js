import React from "react";
import "./Link.css";

export const Link = ({ pathData = "", ...otherProps }) => {
  return <path d={pathData} {...otherProps} />;
};
