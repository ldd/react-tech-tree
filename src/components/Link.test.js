import React from "react";
import renderer from "react-test-renderer";
import { Link } from "./Link";

describe("React Component [Link]", () => {
  it("can create empty Link", () => {
    const component = renderer.create(<Link />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("can create Link with some props", () => {
    const className = "someClass";
    const someProp = "@value";
    const pathData = "M 0 0 L 0 0";
    const component = renderer.create(
      <Link className={className} someProp={someProp} pathData={pathData} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
