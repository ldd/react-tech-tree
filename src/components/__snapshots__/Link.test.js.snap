// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React Component [Link] can create Link with some props 1`] = `
<path
  className="someClass Link"
  d="M 0 0 L 0 0"
  someProp="@value"
/>
`;

exports[`React Component [Link] can create empty Link 1`] = `
<path
  className=" Link"
  d=""
/>
`;
